pac4j-oauth/src/main/java/org/pac4j/oauth/profile/figshare/FigShareProfileDefinition.java

```diff
@@ -0,0 +1,53 @@
+package org.pac4j.oauth.profile.figshare;
+
+import org.pac4j.oauth.profile.JsonHelper;
+import org.pac4j.oauth.profile.definition.OAuth20ProfileDefinition;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+
+import static org.pac4j.core.profile.AttributeLocation.PROFILE_ATTRIBUTE;
+
+import org.pac4j.core.profile.ProfileHelper;
+import org.pac4j.core.profile.converter.Converters;
+import org.pac4j.oauth.config.OAuth20Configuration;
+
+/**
+ * This class is the FigShare profile definition
+ *
+ * @author Vassilis Virvilis
+ * @since 3.8.0
+ */
+public class FigShareProfileDefinition extends OAuth20ProfileDefinition<FigShareProfile, OAuth20Configuration> {
+    public static final String LAST_NAME = "last_name";
+
+    public FigShareProfileDefinition() {
+        super(x -> new FigShareProfile());
+        primary(LAST_NAME, Converters.STRING);
+    }
+
+    @Override
+    public String getProfileUrl(OAuth2AccessToken accessToken, OAuth20Configuration configuration) {
+        return "https://api.figshare.com/v2/account";
+    }
+
+    @Override
+    public FigShareProfile extractUserProfile(final String body) {
+        final FigShareProfile profile = newProfile();
+        final JsonNode json = JsonHelper.getFirstNode(body);
+        if (json != null) {
+            if (getProfileId() != null) {
+                profile.setId(ProfileHelper.sanitizeIdentifier(profile, JsonHelper.getElement(json, getProfileId())));
+            }
+            for (final String attribute : getPrimaryAttributes()) {
+                convertAndAdd(profile, PROFILE_ATTRIBUTE, attribute, JsonHelper.getElement(json, attribute));
+            }
+            for (final String attribute : getSecondaryAttributes()) {
```

Comment:
You don't have defined secondary attributes, do you? -> https://github.com/pac4j/pac4j/pull/1382/files#diff-cacc8a71e4c5a4fe69b0016e9309420dR26